File: services/typescript/fastify/src/routes/router.ts
import type { FastifyInstance } from "fastify";
import type { State } from "../state.js";
import {
  getHandler,
  postHandler,
  patchHandler,
  deleteHandler,
  putHandler,
  invalidMethodHandler
} from "./handlers/index.js";

export function registerRoutes(app: FastifyInstance, state: State) {
  app.post("/", postHandler(state));
  app.route({
    method: ["GET", "HEAD", "PUT", "PATCH", "DELETE", "OPTIONS", "TRACE"],
    url: "/",
    handler: invalidMethodHandler
  });

  app.get("/:id", getHandler(state));
  app.patch("/:id", patchHandler(state));
  app.delete("/:id", deleteHandler(state));
  app.put("/:id", putHandler(state));
  app.route({
    method: ["POST", "OPTIONS", "TRACE"],
    url: "/:id",
    handler: invalidMethodHandler
  });
}

File: services/python/flask/src/routes/router.py
from flask import Blueprint, request
from routes.errors import send_error, AppErrors
from routes.handlers import (
    post_handler,
    get_handler,
    delete_handler,
    put_handler,
    patch_handler,
)

bp = Blueprint("flask", __name__)


def register_routes():
    bp.before_request(reject_head)

    bp.post("/")(post_handler)
    bp.get("/<id>")(get_handler)
    bp.delete("/<id>")(delete_handler)
    bp.put("/<id>")(put_handler)
    bp.patch("/<id>")(patch_handler)


def reject_head():
    if request.method == "HEAD":
        return send_error(AppErrors.InvalidMethod)
    return None
